sklearn/feature_extraction/text.py

```diff
@@ -864,6 +876,10 @@ def transform(self, raw_documents):
         X : sparse matrix, [n_samples, n_features]
             Document-term matrix.
         """
+        if isinstance(raw_documents, str):
+            raise ValueError(
+                "iterable over raw text documents expected, string object given")
```

Comment:
i find it odd that you flip between "received" and "given" ;) maybe choose one and stick to it?